src/components/pages/UpdateFarmer/UpdateFarmer.js

```diff
@@ -0,0 +1,184 @@
+import React, { useState } from 'react';
+import Input from '../../common/Input/Input';
+import * as form from '../../common/Input/addFarmerData';
+import Button from '../../common/Button/StyledButton';
+import styled from 'styled-components';
+import { updateFarmerHandler } from '../../../utils/handlers/farmerHandlers';
+import { getToken } from '../../../utils/handlers/authenticationHandlers';
+import withRestrictedAccess from '../../hoc/withRestrictedAccess';
+
+const UpdateFarmer = ({ location, history, appStateShouldUpdate }) => {
+
+  // Prevents errors when location state is empty
+  const { farmer: farmerData } = location.state || {};
+
+  const hydrateFormInputValues = () => {
+    let hydratedFormInputs = {};
+
+    // Deep copy of form input data objects
+    const formInputData = JSON.parse(JSON.stringify(form));
+
+    for (const inputSection in formInputData) {
+      const inputSectionData = formInputData[inputSection];
+
+      for (const input in inputSectionData) {
+        inputSectionData[input].value = farmerData[inputSection][input];
+        if ('selected' in inputSectionData[input]) {
+          inputSectionData[input].selected = farmerData[inputSection][input];
+        }
+        hydratedFormInputs = {
+          ...formInputData
+        };
+      }
+    }
+
+    return hydratedFormInputs;
+  }
+
+  const [formElementsState, setFormElementsState] = useState(hydrateFormInputValues());
+
+  const [stateToggle, setStateToggle] = useState({
+    personalInfoToggle: false,
+    familyInfoToggle: true,
+    guarantorToggle: true,
+    farmInfoToggle: true,
+  });
+
+  const onChangeHandler = (e, data) => {
+    const { name, value, type } = e.target
+    const newData = { ...formElementsState[data] }
+    const newEntry = { ...newData[name] }
+    if (type === 'checkbox') {
+      if (newEntry.selected.indexOf(value) > -1) {
+        newEntry.selected = newEntry.selected.filter(s => s !== value)
+      } else {
+        newEntry.selected = [...newEntry.selected, value]
+      }
+    } else {
+      newEntry.value = value
+    }
+    newData[name] = newEntry
+    setFormElementsState({ ...formElementsState, [data]: newData })
+  }
+  const toggleHandler = data => {
```

Comment:
I guess you made mention of planning to change the variable name

Reply:
Yes, I will make those changes in a later PR